Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WEB_AUTHN)
+
+#include <WebCore/CredentialRequestCoordinatorClient.h>
+#include <WebCore/FrameIdentifier.h>
+#include <wtf/WeakPtr.h>
+
+namespace WebKit {
+
+class WebPage;
+
+class DigitalCredentialsCoordinator final : public WebCore::CredentialRequestCoordinatorClient {
+public:
+    explicit DigitalCredentialsCoordinator(WebPage&);
+
+private:
+    // WebCore::CredentialRequestCoordinatorClient
+    void requestDigitalCredential(const WebCore::LocalFrame&, const WebCore::DigitalCredentialRequestOptions&, WebCore::DigitalCredentialRequestCompletionHandler&&);
+    void cancel(CompletionHandler<void()>&&) final;
+
+    WeakPtr<WebKit::WebPage> m_webPage;
```

Comment:
Once this object gets constructed, we should look at the ownership model and see if m_webPage can actually be null. My bet is that it can't and we should be using a WeakRef instead of a WeakPtr.